src/models/Events/BenchmarkEvent.test.ts

```diff
@@ -0,0 +1,160 @@
+				id: 'urn:uuid:a84dc6e1-dbde-4da2-a1c8-1415ce19c5eb',
+				type: 'Assessment',
+				subject: 'Reading',
+			},
+			assignee: {
+				id: 'urn:uuid:ec3b84f0-ca55-4b0d-8ffd-4b1cdf5ab9ae',
+				type: 'Student',
+				gradeLevel: 5,
+				otherIdentifiers: [
+					{
+						type: 'SystemIdentifier',
+						identifierType: 'SystemId',
+						identifier: '43c29652-9b24-41bd-af3f-57648672432f',
+						source: {
+							type: 'SoftwareApplication',
+							id: 'https://whatever.com',
+						},
+					},
+				],
+			},
+		},
+		generated: {
+			id: 'urn:uuid:e8d7d46e-edcf-40a1-b31d-b1461cb483c2',
+			type: 'Score',
+			academicTerm: 'SY2021-2022',
+			scoreGiven: 140.0,
+			extensions: {
+				whateverStudent: {
+					id: '43c29652-9b24-41bd-af3f-57648672432f',
+					firstName: 'Foo',
+					lastName: 'Bar',
+					grade: 5,
+				},
+				whateverTest: {
+					id: 'a84dc6e1-dbde-4da2-a1c8-1415ce19c5eb',
+					name: 'Genius Test',
+					timestamp: '2020-01-01T12:00:00Z',
+					subject: 'Language Arts',
+				},
+				whateverResult: {
+					id: 'e8d7d46e-edcf-40a1-b31d-b1461cb483c2',
+					score: 140,
+					answers: ['Person', 'Woman', 'Man', 'Camera', 'TV'],
+				},
+				otherGibberish1: 'Hooplah',
+				otherGibberish2: 12345,
+			},
+		},
+	};
+
+	it('OK', () => {
+		const payload = {
+			whateverStudent: {
+				id: '43c29652-9b24-41bd-af3f-57648672432f',
+				firstName: 'Foo',
+				lastName: 'Bar',
+				grade: 5,
+			},
+			whateverTest: {
+				id: 'a84dc6e1-dbde-4da2-a1c8-1415ce19c5eb',
+				name: 'Genius Test',
+				timestamp: '2020-01-01T12:00:00Z',
+				subject: 'Language Arts',
+			},
+			whateverResult: {
+				id: 'e8d7d46e-edcf-40a1-b31d-b1461cb483c2',
+				score: 140,
+				answers: ['Person', 'Woman', 'Man', 'Camera', 'TV'],
+			},
+			otherGibberish1: 'Hooplah',
+			otherGibberish2: 12345,
+		};
+
+		// Events should only utilize known WNE IDs, not external IDs
+		const wneSubject = 'Reading';
+		const wneStudent = {
+			Id: 'ec3b84f0-ca55-4b0d-8ffd-4b1cdf5ab9ae',
+			GradeLevel: 5,
+		};
+
+		const model = createBenchmarkEvent({
+			actor: createSoftwareApplication({ id: 'https://whatever.com' }),
+			object: createBenchmarkEventAttempt({
+				id: Caliper.uuid(),
+				dateCreated: payload.whateverTest.timestamp,
+				assignable: createBenchmarkEventAssessment({
+					id: Caliper.uuid(payload.whateverTest.id),
+					subject: wneSubject,
+				}),
+				assignee: createBenchmarkEventStudent({
+					id: Caliper.uuid(wneStudent.Id),
+					gradeLevel: wneStudent.GradeLevel,
+					otherIdentifiers: [
+						createSystemIdentifier({
+							identifierType: SystemIdentifierType.SystemId,
+							sourceUrl: 'https://whatever.com',
+							identifier: payload.whateverStudent.id,
+						}),
+					],
+				}),
+			}),
+			generated: createBenchmarkEventScore({
+				id: Caliper.uuid(payload.whateverResult.id),
+				academicTerm: 'SY2021-2022',
+				scoreGiven: payload.whateverResult.score,
+				extensions: payload,
+			}),
+		});
+		validate(model);
```

Comment:
Works either way, so not a big deal. I might wrap these `validate` calls with an `expect` because it gives a slightly smaller stack trace to the TeamCity reporter. ¯\\\_(ツ)\_/¯
```suggestion
		expect(() => validate(model)).not.toThrow();
```